### PR TITLE
Issue 5806: Update HEADER

### LINKED ISSUE #5806: Update HEADER file
**Describe the bug**
The license HEADER needs to be updated to reflect the copyright change.

**Additional context**
With the move to CNCF, we need to update the HEADER file.

### PR DESCRIPTION
**Pull request by fpj**
**Change log description**  
* Updates HEADER file to reflect copyright changes due to the move to CNCF.
* Updates checkstyle to validate the correct copyright.

**Purpose of the change**  
Fixes #5806 

**What the code does**  
There is no code change.

**How to verify it**  
New pull requests with the old or a different copyright statement in the license header should fail on checkstyle.

### CODE REVIEW

File: HEADER [hunk 1]
```diff
@@ -1,7 +1,14 @@
-Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+Copyright 2021 Pravega Authors.
```
Comment (derekm) [inline HEADER:1]: As far as I know, we are still keeping year relegated to NOTICE file. NOTICE file was updated by Igor previously.

**Review by derekm — CHANGES_REQUESTED** [review]

**Comment by derekm** [comment]
We also have Issue #5697

File: HEADER (hunk 1 above)
Comment (fpj) [reply HEADER:1]: That's correct, I'll remove it.

**Review by derekm — APPROVED** [review]

File: checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
   <!-- header -->
   <module name="RegexpHeader">
-    <property name="header" value="/\*\*\n * Copyright \(c\) Dell Inc\.\, or its subsidiaries\. All Rights Reserved\."/>
+    <property name="header" value="/\*\*\n * Copyright \(c\) Pravega Authors. All Rights Reserved\."/>
```
Comment (derekm) [inline checkstyle/checkstyle.xml:12]: Updating the rule with this regex requires we proactively update all file headers throughout the repo. Do we do this sweep change in this PR? Or do we allow the regex to accept both headers?

**Review by derekm — CHANGES_REQUESTED** [review]
We either need to update Checkstyle `RegexpHeader` to accept both headers, or we need to make a sweeping change to update all file headers in the repo.

File: HEADER [hunk 3]
```diff
@@ -1,7 +1,14 @@
-Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+Copyright 2021 The Pravega Authors.
```
Comment (derekm) [inline HEADER:1]: remove year

File: NOTICE [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2021 Pravega Authors.
+Copyright (c) 2021 The Pravega Authors.
```
Comment (derekm) [inline NOTICE:1]: remove "The "

**Review by derekm — CHANGES_REQUESTED** [review]
stray year and stray "The"

**Review by derekm — CHANGES_REQUESTED** [review]
Also -- As in the NOTICE file, we are advised by CNCF policy to not remove old copyright statements, but only add to them. In other words, existing in-place headers should be:
```
/**
 * Copyright Pravega Authors.
 * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
```
or the like.

**Review by derekm — APPROVED** [review]
Expressed permission to change Dell copyright statements given by Sean Evans of Legal:
>"Confirming both that we can change the Dell copyright statements and we cannot change the copyright statements for non-Dell authors."

**Pull request merged**